apps/frontend/src/lib/components/ChatMessages.svelte

```diff
@@ -45,8 +67,12 @@
 		</div>
 	{/each}
 	{#if generationError && onRetryError}
-		<ChatErrorMessage error={generationError} onRetry={onRetryError} />
+		<div {@attach scrollToMe()} transition:fly={{ y: 20, duration: 800 }}>
+			<ChatErrorMessage error={generationError} onRetry={onRetryError} />
+		</div>
 	{:else if !finalAnswerStarted}
-		<ChatWaiting />
+		<div {@attach scrollToMe()} transition:fly={{ y: 20, duration: 800 }}>
+			<ChatWaiting />
+		</div>
```

Comment:
This seems like unnecessary code duplication. Why not add the `<div>` _around_ the conditional `{#if`?

That having been said, we _will_ be moving towards an infinite scroll pattern for full history, preventing performance problems on exceedingly longer conversations.